app/server/models/user.py

```diff
@@ -254,6 +257,19 @@ def tfa_url(self):
                 current_app.config.get('DEPLOYMENT_NAME'))
         )
 
+    @hybrid_property
+    def full_location(self):
+        if len(self._full_location) == 0:
+            return None
+        return self._full_location[0]
+
+    @full_location.setter
+    def full_location(self, location):
+        if len(self._full_location) == 0:
+            self._full_location.append(location)
+        else:
+            self._full_location[0] = location
+
     @hybrid_property
     def location(self):
```

Comment:
Yes, definitely. However, there is already another PR that introduces updating the old location stucture. I would like then to do this in two passes, where GUI is tied to new location input next. It may involve a map picker etc, so a bit of work in quite a different domain.

Furthermore it seems we need an "inbox" of new suggested location data from surveys. I think it should be done as an append log, but it seems like a different task, and while it's not there the old fields in the user table seem to suffice.